 contrib/CMakeLists.txt | 12 +++++++-----
 gsl_version.h          | 26 ++++++++++++++++++++++++++
 2 files changed, 33 insertions(+), 5 deletions(-)

diff --git a/contrib/CMakeLists.txt b/contrib/CMakeLists.txt
index 0574403..e5ec979 100644
--- a/contrib/CMakeLists.txt
+++ b/contrib/CMakeLists.txt
@@ -9,7 +9,7 @@ set(PACKAGE_TARNAME ${PACKAGE})
 set(PACKAGE_BUGREPORT "")
 set(PACKAGE_URL "")
 set(LT_OBJDIR ".libs/")
-
+set(DO_TESTS "True")
 # The following part of config.h is hard to derive from configure.ac.
 string(REGEX MATCHALL "[^\n]*\n" CONFIG
 "/* Define if you have inline with C99 behavior */
@@ -582,13 +582,15 @@ foreach (dir "." ${dirs})
         set(GSL_SOURCES ${GSL_SOURCES} ${SOURCES})
       endif ()
     elseif (line MATCHES "^test.*_SOURCES[ \t]*=")
-      get_sources(${dir} "${line}" SOURCES)
-      string(REGEX REPLACE "(.*)_SOURCES.*" "\\1" suffix ${line})
-      add_gsl_test("${dir}_${suffix}" ${SOURCES})
+      if(${DO_TESTS} EQUAL "True")
+        get_sources(${dir} "${line}" SOURCES)
+        string(REGEX REPLACE "(.*)_SOURCES.*" "\\1" suffix ${line})
+        add_gsl_test("${dir}_${suffix}" ${SOURCES})
+      endif()
     endif ()
   endforeach ()
 endforeach ()
-
+message(STATUS ${DO_TESTS})
 add_library(gsl ${GSL_SOURCES})
 add_dependencies(gsl copy-headers)
 
diff --git a/gsl_version.h b/gsl_version.h
new file mode 100644
index 0000000..6fd5ede
--- /dev/null
+++ b/gsl_version.h
@@ -0,0 +1,26 @@
+#ifndef __GSL_VERSION_H__
+#define __GSL_VERSION_H__
+
+#include <gsl/gsl_types.h>
+
+#undef __BEGIN_DECLS
+#undef __END_DECLS
+#ifdef __cplusplus
+# define __BEGIN_DECLS extern "C" {
+# define __END_DECLS }
+#else
+# define __BEGIN_DECLS /* empty */
+# define __END_DECLS /* empty */
+#endif
+__BEGIN_DECLS
+
+
+#define GSL_VERSION "2.1"
+#define GSL_MAJOR_VERSION 2
+#define GSL_MINOR_VERSION 1
+
+GSL_VAR const char * gsl_version;
+
+__END_DECLS
+
+#endif /* __GSL_VERSION_H__ */
